manifests/agent/service/cron.pp

```diff
@@ -1,7 +1,7 @@
 class puppet::agent::service::cron (
   Boolean $enabled = false,
 ) {
-  unless 'cron' in $::puppet::unavailable_runmodes {
+  unless 'unmanaged' in $::puppet::runmode or 'cron' in $::puppet::unavailable_runmodes {
```

Comment:
runmode is a string, so you can compare it with `$::puppet::runmode == 'unmanaged'`